cli/src/lockfiles/python.rs

```diff
@@ -130,18 +136,35 @@ struct PoetryLock {
 struct Package {
     name: String,
     version: String,
+    source: Option<PackageSource>,
 }
 
 impl From<Package> for PackageDescriptor {
     fn from(package: Package) -> Self {
+        let version = package
+            .source
+            .and_then(|source| {
+                let reference = source.resolved_reference.as_ref();
+                reference.map(|reference| format!("{}#{}", source.url, reference))
```

Comment:
There is still some uncertainty...but the current thought is that the format for the version on this line is not quite what the `request-manager` expects. Feeding the [`poetry.lock` file from `vulnerabilities-admin`](https://github.com/phylum-dev/vulnerabilities-admin/blob/poetry_month/poetry.lock) results in the following error...which appears to be due to failing to match any of the version regex's in request-manager...specifically [this one](https://github.com/phylum-dev/request-manager/blob/bb1d4bfecabb63a8aba79ed86df91af31dd1fb82/src/models.py#L480).

```sh
❯ ./phylum analyze -l poetry_type_test_4 -F poetry.lock
[2022-04-19T17:16:34Z ERROR phylum] Execution failed: Improperly formatted creation request
❗ Error: Execution failed caused by: Improperly formatted creation request
```

My regex foo is limited, but perhaps @ejortega can assist.

Reply:
The regex definitely doesn't match `https` URLs. I'm not sure any format that is supported in `poetry.lock` is also supported by this regex.

Reply:
I'm testing out some changes with request-manager and api

Reply:
The URLs work correctly now.